### PR TITLE
Unscented transform, predict, and update implementation; Tests for KF/UKF

### PR DESCRIPTION
**Pull request by ArthurAllshire**

### CODE REVIEW

File: utilities/kalman.py [hunk 1]
```diff
@@ -10,19 +10,42 @@ def m_dot(*args):
...
         self.Q = Q
         self.R = R
         self.history=deque(iterable=[[self.x_hat, self.P]], maxlen=history_len)
+        self.f = f
+        self.h = h
+
+        def subtraction_supported_residual(a,b):
+            return a-b
+        self.residual_x = residual_x if residual_x else subtraction_supported_residual
+        self.residual_z = residual_z if residual_z else subtraction_supported_residual
```
Comment (auscompgeek) [inline utilities/kalman.py]: protip: the `or` operator is a thing

File: utilities/kalman.py [hunk 2]
```diff
@@ -79,7 +102,135 @@ def update(self, z, H, R=None):
...
+        if f is None:
+            if self.f is None:
+                print("WARNING: Kalman unscented_predict requires a state transition function")
+                return
+            f = self.f
+
+        if Q is None:
+            if self.Q is None:
+                print("WARNING- Kalman unscented_predict requires a process noise covariance matrix")
+                return
```
Comment (auscompgeek) [inline utilities/kalman.py]: These _really_ should be exceptions.

File: utilities/kalman.py (hunk 1 above)
Comment (ArthurAllshire) [inline utilities/kalman.py]: what do you mean?

Comment (auscompgeek) [inline utilities/kalman.py]: you should look up what `or` does in python

Comment (ArthurAllshire) [inline utilities/kalman.py]: oh, just realized what you mean...

**Comment by ArthurAllshire** [comment]
> It'd be nice if there were tests for the Kalman filter…
agreed. I wonder what the best way to do this is? I'm thinking maybe testing for steady state filter convergence or something similar?

**Comment by ArthurAllshire** [comment]
Added convergence tests.

**Pull request merged**